application.go

```diff
@@ -56,43 +56,44 @@ type Port struct {
-	Executor                   *string              `json:"executor,omitempty"`
-	HealthChecks               *[]HealthCheck       `json:"healthChecks,omitempty"`
-	Instances                  *int                 `json:"instances,omitempty"`
-	Mem                        *float64             `json:"mem,omitempty"`
-	Tasks                      []*Task              `json:"tasks,omitempty"`
-	Ports                      []int                `json:"ports"`
-	PortDefinitions            *[]PortDefinition    `json:"portDefinitions,omitempty"`
-	RequirePorts               *bool                `json:"requirePorts,omitempty"`
-	BackoffSeconds             *float64             `json:"backoffSeconds,omitempty"`
-	BackoffFactor              *float64             `json:"backoffFactor,omitempty"`
-	MaxLaunchDelaySeconds      *float64             `json:"maxLaunchDelaySeconds,omitempty"`
-	TaskKillGracePeriodSeconds *float64             `json:"taskKillGracePeriodSeconds,omitempty"`
-	Deployments                []map[string]string  `json:"deployments,omitempty"`
-	Dependencies               []string             `json:"dependencies"`
-	TasksRunning               int                  `json:"tasksRunning,omitempty"`
-	TasksStaged                int                  `json:"tasksStaged,omitempty"`
-	TasksHealthy               int                  `json:"tasksHealthy,omitempty"`
-	TasksUnhealthy             int                  `json:"tasksUnhealthy,omitempty"`
-	TaskStats                  map[string]TaskStats `json:"taskStats,omitempty"`
-	User                       string               `json:"user,omitempty"`
-	UpgradeStrategy            *UpgradeStrategy     `json:"upgradeStrategy,omitempty"`
-	Uris                       *[]string            `json:"uris,omitempty"`
-	Version                    string               `json:"version,omitempty"`
-	VersionInfo                *VersionInfo         `json:"versionInfo,omitempty"`
-	Labels                     *map[string]string   `json:"labels,omitempty"`
-	AcceptedResourceRoles      []string             `json:"acceptedResourceRoles,omitempty"`
-	LastTaskFailure            *LastTaskFailure     `json:"lastTaskFailure,omitempty"`
-	Fetch                      *[]Fetch             `json:"fetch,omitempty"`
-	IPAddressPerTask           *IPAddressPerTask    `json:"ipAddress,omitempty"`
+	ID                         string                  `json:"id,omitempty"`
+	Cmd                        *string                 `json:"cmd,omitempty"`
+	Args                       *[]string               `json:"args,omitempty"`
+	Constraints                *[][]string             `json:"constraints,omitempty"`
+	Container                  *Container              `json:"container,omitempty"`
+	CPUs                       float64                 `json:"cpus,omitempty"`
+	Disk                       *float64                `json:"disk,omitempty"`
+	Env                        *map[string]interface{} `json:"env,omitempty"`
```

Comment:
I'm not happy that we are using `interface` now. Is there a way to avoid it and bring back type safety?